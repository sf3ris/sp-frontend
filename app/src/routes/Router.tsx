import React from 'react';
import { routes, IRoute } from './routes';
import { Route, Switch, Redirect } from 'react-router-dom';
import ProfileContainer from '../container/ProfileContainer';

const Router : React.FC<{}> = props => {

    const renderRoute = ( route : IRoute, index : number ) => <Route key={index} path={route.path} component={route.component} />

    return (

        <Switch>

            {routes.map(renderRoute)}

            <Route path="/profile" exact component={ProfileContainer} />
            <Redirect to='/home' />

        </Switch>

    )

}

export default Router;