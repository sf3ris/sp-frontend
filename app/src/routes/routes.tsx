import React from 'react';
import { IconProp } from '@fortawesome/fontawesome-svg-core'
import { faBookmark, faCalendar,faUser } from '@fortawesome/free-regular-svg-icons';
import { faHome, faCog } from '@fortawesome/free-solid-svg-icons';
import HomeContainer from '../container/HomeContainer';
import ProfileContainer from '../container/ProfileContainer';

export interface IRoute {
    name : string;
    path : string;
    component? : React.FC;
    icon : IconProp;
    hideFromMenu? : boolean;
}

export const routes : IRoute[] = [
    {
        name: 'Home',
        path: '/home',
        icon : faHome,
        component: HomeContainer
    },
    {
        name: 'Bookmars',
        path: '/bookmarks',
        icon: faBookmark,
    },
    {
        name: 'Calendar',
        path: '/calendar',
        icon: faCalendar
    },
    {
        name: 'Settings',
        path: '/settings',
        icon: faCog
    },
    {
        name: 'Profile',
        path: '/profile',
        icon: faUser,
        hideFromMenu: true,
        component: ProfileContainer
    }
]