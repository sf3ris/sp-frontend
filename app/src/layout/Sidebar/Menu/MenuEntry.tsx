import React from 'react';
import { IRoute } from '../../../routes/routes';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Link } from 'react-router-dom';

interface IMenuEntryProps {
    route : IRoute;
    collapsed : boolean;
}

const MenuEntry : React.FC<IMenuEntryProps> = props => {
 
    return (

        
        <li>
            <Link to={props.route.path}> 
                <FontAwesomeIcon icon={props.route.icon}/> {props.collapsed ? null : <span>{props.route.name}</span> }
            </Link>
        </li>

    )

}

export default MenuEntry;