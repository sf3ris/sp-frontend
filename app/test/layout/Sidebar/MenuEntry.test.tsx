import React from 'react';
import MenuEntry from '../../../src/layout/Sidebar/Menu/MenuEntry';
import { IRoute } from '../../../src/routes/routes';
import { shallow } from 'enzyme';

describe('Menu Entry component test', () => {

    it('Should match snapshot', () => {

        const routeMock = { name: 'route_1', path: '/route_1' } as IRoute;

        const wrapper = shallow( <MenuEntry collapsed={false} route={routeMock} /> )

        expect(wrapper.debug()).toMatchSnapshot();

    });

    it('Should render text', () => {

        const routeMock = { name: 'route_1', path: '/route_1' } as IRoute;

        const wrapper = shallow( <MenuEntry collapsed={false} route={routeMock} /> )

        expect(wrapper.find('span').length).toBe(1);

    });

    it('Should not render text', () => {

        const routeMock = { name: 'route_1', path: '/route_1' } as IRoute;

        const wrapper = shallow( <MenuEntry collapsed={true} route={routeMock} /> )

        expect(wrapper.find('span').length).toBe(0);

    });
    
})
