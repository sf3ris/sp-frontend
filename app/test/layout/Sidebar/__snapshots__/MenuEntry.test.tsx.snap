// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Entry component test Should match snapshot 1`] = `
"<li>
  <Link to=\\"/route_1\\">
    <FontAwesomeIcon icon={{...}} border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false} />
     
    <span>
      route_1
    </span>
  </Link>
</li>"
`;
